packages/mermaid/src/diagrams/gantt/ganttDb.js

```diff
@@ -343,6 +343,35 @@ const parseDuration = function (str) {
 const getEndDate = function (prevTime, dateFormat, str, inclusive = false) {
   str = str.trim();
 
+  // Test for until
+  const re = /^until\s+([\d\w- ]+)/;
+  const untilStatement = re.exec(str.trim());
+
+  if (untilStatement !== null) {
+    // check all until ids and take the earliest
+    let earliestStartingTask = null;
+    untilStatement[1].split(' ').forEach(function (id) {
```

Comment:
Use named capture groups instead of indexes. 

```js
  const untilStatement = /^until\s+(?<ids>[\d\w- ]+)/.exec(str);
  if (untilStatement) {
    // check all until ids and take the earliest
    let earliestStartingTask = null;
    for (const id of untilStatement.groups.ids.split(' ')) {
```